### PR TITLE
config: rename NewGrpcMuxImpl -> GrpcMuxImpl

### PR DESCRIPTION
**Pull request by wgallagher**
For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Description: rename NewGrpcMuxImpl -> GrpcMuxImpl
Risk Level: low
Testing: n/a
Docs Changes: n/a
Release Notes: n/a
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

**Comment by mattklein123** [comment]
Thanks @wgallagher can you fix format and DCO (see contributor guide for setting things up)?

/wait

**Comment by mattklein123** [comment]
Putting into waiting while we deal with the revert.

/wait

**Comment by mattklein123** [comment]
@wgallagher please fix format and undraft when ready for review. Thank you.

/wait

**Comment by mattklein123** [comment]
Please fix DCO.

/wait

File: source/common/config/new_grpc_mux_impl.cc [hunk 1]
```diff
@@ -10,17 +10,17 @@
 namespace Envoy {
 namespace Config {
 
-NewGrpcMuxImpl::NewGrpcMuxImpl(std::unique_ptr<SubscriptionStateFactory> subscription_state_factory,
```
Comment (mattklein123) [inline source/common/config/new_grpc_mux_impl.cc]: Can you rename the files also to get rid of new? (Here and in all the other files).

Comment (wgallagher) [reply source/common/config/new_grpc_mux_impl.cc]: +1

File: test/common/config/BUILD [hunk 2]
```diff
@@ -94,28 +94,6 @@ envoy_cc_test(
     ],
 )
 
-envoy_cc_test(
```
Comment (mattklein123) [inline test/common/config/BUILD]: Did we lose coverage here somehow?

Comment (wgallagher) [reply test/common/config/BUILD]: this class contained a single test which seemed to be duplicated in grpc_mux_impl_test.cc

Comment (wgallagher) [reply test/common/config/BUILD]: Specifically, TEST_F(GrpcMuxImplTest, DiscoveryResponseNonexistentSub) which does everything this test does and adds a few additional checks.

**Pull request merged**